### PR TITLE
ADD: Created BlueAddressInput

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

File: screen/send/details.js [hunk 1]
```diff
@@ -82,11 +81,15 @@ export default class SendDetails extends Component {
...
-    EV(EV.enum.CREATE_TRANSACTION_NEW_DESTINATION_ADDRESS, data => {
-      this.setState(
-        { isLoading: false },
-        () => {
+  processAddressData = data => {
+    this.props.navigation.goBack(null);
+    this.setState(
+      { isLoading: true },
+      () => {
+        console.warn(data);
```
Comment (Overtorment) [inline screen/send/details.js]: remove warn

**Comment by Overtorment** [comment]
example bip21 
image

File: screen/send/scanQrAddress.js [hunk 2]
```diff
@@ -70,7 +74,8 @@ export default class CameraExample extends React.Component {
 
 CameraExample.propTypes = {
   navigation: PropTypes.shape({
-    goBack: PropTypes.function,
-    dismiss: PropTypes.function,
+    goBack: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/send/scanQrAddress.js]: func is the constant for the PropTypes type. function is an constructor for a JS function.

File: NavigationService.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+function navigate(routeName, params) {
+  _navigator.dispatch(
+    NavigationActions.navigate({
+      routeName,
+      params,
+    }),
+  );
+}
+
+// add other navigation functions that you need and export them
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply NavigationService.js]: How come?

File: NavigationService.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// NavigationService.js
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply NavigationService.js]: How come?

File: android/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">BlueWallet</string>
+    <string name="app_name">Blue Wallet</string>
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply android/app/src/main/res/values/strings.xml]: How do we wanna call the app? BlueWallet or Blue Wallet?

File: events.js [hunk 6]
```diff
@@ -36,10 +36,6 @@ EV.enum = {
   // changed (usually for current wallet)
   REMOTE_TRANSACTIONS_COUNT_CHANGED: 'REMOTE_TRANSACTIONS_COUNT_CHANGED',
 
-  // emitted when QR scanner scanned address that should be used in CREATE TRANSACTION screen
-  // thus, previous screen (CREATE TRANSACTION screen) will update it's input content
-  CREATE_TRANSACTION_NEW_DESTINATION_ADDRESS: 'CREATE_TRANSACTION_NEW_DESTINATION_ADDRESS',
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply events.js]: Thats correct.

File: ios/BlueWallet/Info.plist [hunk 7]
```diff
@@ -5,7 +5,7 @@
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
 	<key>CFBundleDisplayName</key>
-	<string>BlueWallet</string>
+	<string>Blue Wallet</string>
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply ios/BlueWallet/Info.plist]: How do we wanna call the app? BlueWallet or Blue Wallet?

File: NavigationService.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Overtorment) [reply NavigationService.js]: i mean, comment string. what it does?

File: android/app/src/main/res/values/strings.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Overtorment) [reply android/app/src/main/res/values/strings.xml]: @ncoelho ^^^

File: screen/send/scanQrAddress.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Overtorment) [reply screen/send/scanQrAddress.js]: this does not make any sense to me, but Im no JS dev so why not  ¯\_(ツ)_/¯

**Comment by Overtorment** [comment]
Problem 2: after scan qr screen goes back too far

btc wallet send - wtf - peek 2019-01-25 20-05

File: screen/lnd/scanLndInvoice.js [hunk 8]
```diff
@@ -29,7 +35,7 @@ export default class ScanLndInvoice extends React.Component {
 
     if (!BlueApp.getWallets().some(item => item.type === LightningCustodianWallet.type)) {
       alert('Before paying a Lightning invoice, you must first add a Lightning wallet.');
-      props.navigation.dismiss();
+      // props.navigation.dismiss();
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply screen/lnd/scanLndInvoice.js]: OOps. yeah it was a mistake.

**Comment by marcosrdz** [comment]
> Something wrong with scanning bip21. Either with QR generated, either with scanning it:
> 
> ```
> 01-24 14:06:03.509  4594  4627 I ReactNativeJS: { [Error: Invalid BIP21 URI: 3MbZrHLsxdBmAJnB5CC6nfRsKLwGZ5Cx31?amount=01&label=Test]
> 01-24 14:06:03.509  4594  4627 I ReactNativeJS:   line: 211342,
> 01-24 14:06:03.509  4594  4627 I ReactNativeJS:   column: 123,
> 01-24 14:06:03.509  4594  4627 I ReactNativeJS:   sourceURL: 'http://10.0.2.2:8081/index.delta?platform=android&dev=true&minify=false' }
> 01-24 14:06:03.509  4594  4627 I ReactNativeJS: undefined
> ```
> And while youre here, can you pls take a look why this button has no text?
> 
> image

Fixed.

**Comment by marcosrdz** [comment]
> problem 1
> 
> image

FIXED

**Pull request merged**